src/components/incident/test/IncidentDetails.test.tsx

```diff
@@ -0,0 +1,164 @@
+
+const sourceSystemMock: SourceSystem = {
+    pk: 100,
+    name: 'Test Source System',
+    type: 'Test Type'
+}
+
+const incidentTagsMock: IncidentTag[] = [
+    {added_by: 1, added_time: '2021-06-28 08:29:06', tag: 'Test tag'}
+]
+
+let resolvedIncidentMock: Incident = {
+    level: 1,
+    pk: 1000,
+    start_time: '2021-06-28 08:29:06',
+    end_time: '2021-08-28 08:29:06',
+    stateful: true,
+    details_url: '',
+    description: 'Test resolved (closed) incident',
+    ticket_url: '',
+    open: false,
+    acked: false,
+
+    source: sourceSystemMock,
+    source_incident_id: '1001',
+
+    tags: incidentTagsMock
+}
+
+let openIncidentMock: Incident = {
+    level: 1,
+    pk: 2000,
+    start_time: '2021-06-28 08:29:06',
+    end_time: 'infinity',
+    stateful: true,
+    details_url: '',
+    description: 'Test open incident',
+    ticket_url: '',
+    open: true,
+    acked: false,
+
+    source: sourceSystemMock,
+    source_incident_id: '2001',
+
+    tags: incidentTagsMock
+}
+
+let statelessIncidentMock: Incident = {
+    level: 5,
+    pk: 3000,
+    start_time: '2021-06-28 08:29:06',
+    end_time: undefined,
+    stateful: false,
+    details_url: '',
+    description: 'Test stateless incident',
+    ticket_url: '',
+    open: false,
+    acked: false,
+
+    source: sourceSystemMock,
+    source_incident_id: '3001',
+
+    tags: incidentTagsMock
+}
+
+describe('Incident Details: end time and duration are displayed correctly, depending on incident status',() => {
+    beforeEach(() => {
+        jest.spyOn(console, 'error').mockImplementation(() => {});
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+        jest.resetAllMocks();
+    })
+
+    it("should display both end time and duration (closed incident)", () => {
+        render(
+            <IncidentDetails
+                incident={resolvedIncidentMock}
+                onIncidentChange={onIncidentChange}
+            />
+        )
+
+        // Primary details section is rendered correctly
+        const primaryDetailsContainer = screen.getByTestId(/primary-details-container/i)
+        expect(primaryDetailsContainer).toBeInTheDocument()
+        expect(within(primaryDetailsContainer).getByText(/primary details/i)).toBeInTheDocument() // title
+        expect(within(primaryDetailsContainer).getByRole('list')).toBeInTheDocument()
+        expect(within(primaryDetailsContainer).getAllByRole('listitem').length).toBeGreaterThanOrEqual(6)
+
+        // End time item is displayed
+        const endTimeItem = within(primaryDetailsContainer).getByTitle(/end-time/i)
+        expect(endTimeItem).toBeInTheDocument()
+        if (resolvedIncidentMock.end_time) {
+            expect(within(endTimeItem).getByText(resolvedIncidentMock.end_time)).toBeInTheDocument()
+        }
+
+        // Duration item is displayed
+        const durationItem = within(primaryDetailsContainer).getByTitle(/duration/i)
+        expect(durationItem).toBeInTheDocument()
```

Comment:
above you check if the end_time value itself is rendered. Here you dont check if the duration value is rendered, you only check if there is something within the container with title matching `/duration/i`